ui/DOMSelection.js

```diff
@@ -78,6 +89,19 @@ DOMSelection.Prototype = function() {
     if (sel.isCollapsed()) {
       wRange.setStart(start.container, start.offset);
       wSel.addRange(wRange);
+      var el = document.createElement("span");
+      el.id = 'cursor';
+      wRange.deleteContents();
+      wRange.insertNode(el);
+      var visible = isElementInViewport(el);
+      if ($('#cursor')[0].offsetTop < $(".lt-editors")[0].scrollTop){
+        // scrolling up
+        if (!visible) $(".lt-editors")[0].scrollTop = $('#cursor')[0].offsetTop - 200;
+      } else {
+        // scrolling down
```

Comment:
If there is sufficient text below the cursor, this will make the cursor go past the top edge. I think in the case of scrolling down, we want to set scrollTop to something like `offsetTop - viewport height  + our adjustment`